### ISSUE TITLE
Uncaught error - when ACL feature is turned on, and login credentials not passed 

### ISSUE THREAD

**Issue by prashant-shahi** [dgraph-io/dgraph-js#71]
When I ran the example script against *dgraph cluster* with **ACL** feature turned on, I encountered the following error, which wasn't of much help.  

```
ERROR:  TypeError: Cannot read property 'serializeBinary' of undefined
    at Object.createPayload (/Users/prashant/dgraph-code/dgraph-js/examples/v1.1/node_modules/dgraph-js/lib/types.js:65:105)
    at DgraphClient.<anonymous> (/Users/prashant/dgraph-code/dgraph-js/examples/v1.1/node_modules/dgraph-js/lib/client.js:86:36)
    at step (/Users/prashant/dgraph-code/dgraph-js/examples/v1.1/node_modules/dgraph-js/lib/client.js:32:23)
    at Object.throw (/Users/prashant/dgraph-code/dgraph-js/examples/v1.1/node_modules/dgraph-js/lib/client.js:13:53)
    at rejected (/Users/prashant/dgraph-code/dgraph-js/examples/v1.1/node_modules/dgraph-js/lib/client.js:5:65)
```

Meanwhile, we need to have more tests around ACL as well.

**Comment by prashant-shahi** [comment]
Even when the Dgraph cluster is not properly up and running, or invalid gRPC address is used, the same issue was encountered.

**Comment by prashant-shahi** [comment]
Turns out this issue also happens when an invalid schema is passed.

It will be resolved with the release of `v2.0.2`.
